feature-spec-gen/src/main/java/org/wildfly/galleon/plugin/featurespec/generator/FeatureSpecNode.java

```diff
@@ -161,13 +161,17 @@ private static boolean samePackages(ModelNode standaloneDescr, ModelNode domainD
             }
         }
         final List<ModelNode> packagesDescr = domainDescr.hasDefined("packages") ? domainDescr.get("packages").asList() : Collections.emptyList();
-        if(packagesDescr.size() != standalonePackages.size()) {
+        Set<ModelNode> domainPackages = new HashSet<>(packagesDescr.size());
+        for (ModelNode packageDep : packagesDescr) {
+            domainPackages.add(packageDep);
+        }
+        if(domainPackages.size() != standalonePackages.size()) {
             if(failIfDifferent) {
                 throw new ProvisioningException("Packages dependencies don't match");
```

Comment:
Your change avoids the failure, but  if something goes wrong this message is still pretty useless. So part of the fix for WFGP-199 should be to make a better message.

Reply:
@bstansberry , emmanuel fixed the exception message. LGTM.